core/domain/state_domain.py

```diff
@@ -2052,6 +2333,15 @@ def validate(self) -> None:
                 'Expected needs_update to be a bool, received %s' %
                 self.needs_update)
 
+        # Validate translations.
+        if isinstance(self.translation, list):
```

Comment:
Instead of "isinstance" should we use the data_format field to check the kind of translation?

Reply:
The `data_format` field will always be of type `str` and here I'm using `isinstance` because the translation can be an `str` or `List[str]` as per the type annotations in the `init` function. So I think we can't really check the type from `data_format`.
Thanks

Reply:
To clarify, I meant, can we use the value of data_format to determine what is the format of translation (e.g. HTML, unicode, set_of_normalized_string) instead of using "isinstance"?

Reply:
Oh, I see. Sorry I was thinking something else. Thanks for clearing it up. I have updated the code.
Thanks